packages/flutter/lib/src/cupertino/modal_bottom_sheet.dart

```diff
@@ -0,0 +1,389 @@
+    );
+  }
+
+  @override
+  Widget getPreviousRouteTransition(BuildContext context,
+      Animation<double> secondaryAnimation, Widget child) {
+    return _CupertinoModalTransition(
+      secondaryAnimation: secondaryAnimation,
+      body: child,
+      animationCurve: previousRouteAnimationCurve,
+      topRadius: topRadius,
+      backgroundColor: transitionBackgroundColor ?? CupertinoColors.black,
+    );
+  }
+}
+
+class _CupertinoModalTransition extends StatelessWidget {
+  final Animation<double> secondaryAnimation;
+  final Radius topRadius;
+  final Curve? animationCurve;
+  final Color backgroundColor;
+
+  final Widget body;
+
+  const _CupertinoModalTransition({
+    Key? key,
+    required this.secondaryAnimation,
+    required this.body,
+    required this.topRadius,
+    this.backgroundColor = CupertinoColors.black,
+    this.animationCurve,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final paddingTop = MediaQuery.of(context)?.padding.top ?? 0;
+
+    final phoneWithRoundedCorners = defaultTargetPlatform == TargetPlatform.iOS && paddingTop > 20;
+    final startRoundCorner = phoneWithRoundedCorners ? _deviceFrameCorners : 0;
+
+    final curvedAnimation = CurvedAnimation(
+      parent: secondaryAnimation,
+      curve: animationCurve ?? Curves.easeOut,
+    );
+
+    return AnnotatedRegion<SystemUiOverlayStyle>(
+      value: SystemUiOverlayStyle.light,
+      child: AnimatedBuilder(
+        animation: curvedAnimation,
+        child: body,
+        builder: (context, child) {
+          final progress = curvedAnimation.value;
+          final yOffset = progress * paddingTop;
+          final scale = 1 - progress / 10;
+          final radius = progress == 0
+              ? 0.0
+              : (1 - progress) * startRoundCorner + progress * topRadius.x;
+          return Stack(
+            children: <Widget>[
+              Container(color: backgroundColor),
+              Transform.translate(
+                offset: Offset(0, yOffset),
+                child: Transform.scale(
+                  scale: scale,
+                  alignment: Alignment.topCenter,
+                  child: ClipRRect(
+                      borderRadius: BorderRadius.circular(radius),
+                      child: child),
+                ),
+              ),
+            ],
+          );
+        },
+      ),
+    );
+  }
+}
+
+class _CupertinoScaffold extends InheritedWidget {
+  final AnimationController animation;
+
+  final Radius topRadius;
+
+  @override
+  final Widget child;
+
+  const _CupertinoScaffold(
+      {Key? key,
+      required this.animation,
+      required this.child,
+      required this.topRadius})
+      : super(key: key, child: child);
+
+  @override
+  bool updateShouldNotify(InheritedWidget oldWidget) {
+    return false;
+  }
+}
+
+//  This is can be removed once MaterialPageRoute suppors animated modals
+class CupertinoScaffold extends StatefulWidget {
```

Comment:
I would definitely not re-export something specifically for routes with such a generic name.